gitlab-gcp.yaml

```diff
@@ -0,0 +1,4795 @@
+    release: gitlab-demo
+    heritage: Tiller
+    
+  annotations:
+    
+spec:
+  type: ClusterIP
+  ports:
+    - port: 22
+      targetPort: 2222
+      protocol: TCP
+      name: ssh
+  selector:
+    app: gitlab-shell
+    release: gitlab-demo
+
+---
+# Source: gitlab/charts/gitlab/charts/unicorn/templates/service.yaml
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: gitlab-demo-unicorn
+  namespace: gitlab
+  labels:
+    app: unicorn
+    chart: unicorn-1.8.4
+    release: gitlab-demo
+    heritage: Tiller
+    
+  annotations:
+    
+spec:
+  type: ClusterIP
+  ports:
+    - port: 8080
+      targetPort: 8080
+      protocol: TCP
+      name: unicorn
+    - port: 8181
+      targetPort: 8181
+      protocol: TCP
+      name: workhorse
+  selector:
+    app: unicorn
+    release: gitlab-demo
+
+---
+# Source: gitlab/charts/nginx-ingress/templates/controller-metrics-service.yaml
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: gitlab-demo-nginx-ingress-controller-metrics
+  namespace: gitlab
+  annotations:
+    prometheus.io/port: "10254"
+    prometheus.io/scrape: "true"
+  labels:
+    app: nginx-ingress
+    chart: nginx-ingress-0.30.0-1
+    release: gitlab-demo
+    heritage: Tiller
+    
+    component: "controller"
+spec:
+  clusterIP: ""
+  ports:
+    - name: metrics
+      port: 9913
+      targetPort: metrics
+  selector:
+    app: nginx-ingress
+    component: "controller"
+    release: gitlab-demo
+  type: "ClusterIP"
+
+---
+# Source: gitlab/charts/nginx-ingress/templates/controller-service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: gitlab-demo-nginx-ingress-controller
+  namespace: gitlab
+  labels:
+    component: "controller"
+    app: nginx-ingress
+    chart: nginx-ingress-0.30.0-1
+    release: gitlab-demo
+    heritage: Tiller
+    
+spec:
+  # 0.31.0 gates this on `if .Values.controller.service.clusterIP`,
+  # but that breaks upgrades due to immutability.
+  # See https://github.com/helm/charts/pull/7933
+  clusterIP: ""
+  externalTrafficPolicy: "Local"
+  ports:
+    - name: http
+      port: 80
+      protocol: TCP
+      targetPort: http
+    - name: https
+      port: 443
+      protocol: TCP
+      targetPort: https
+    - name: gitlab-shell
+      port: 22
+      protocol: TCP
+      targetPort: gitlab-shell
+  selector:
+    app: nginx-ingress
+    component: "controller"
+    release: gitlab-demo
+  type: "LoadBalancer"
+
+---
+# Source: gitlab/charts/nginx-ingress/templates/controller-stats-service.yaml
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: gitlab-demo-nginx-ingress-controller-stats
+  namespace: gitlab
+  labels:
+    app: nginx-ingress
+    chart: nginx-ingress-0.30.0-1
+    release: gitlab-demo
+    heritage: Tiller
+    
+    component: "controller"
+spec:
+  clusterIP: ""
+  ports:
+    - name: stats
+      port: 18080
+      targetPort: stats
+  selector:
+    app: nginx-ingress
+    component: "controller"
+    release: gitlab-demo
+  type: "Cl
```

Comment:
what are all these blank entries? are the indicative of a problem? should they be removed?

Reply:
Those are generated by `helm template` and most likely due to optional or "not configured" components. This demo does not modify either the GitLab helm chart nor rendered yaml file.